pkg/agents/team6/config.go

```diff
@@ -213,34 +283,46 @@ func (a *CustomAgent6) Run() {
 	a.addToMemory()
 
 	// Eat if needed/wanted
-	foodTaken, err := a.TakeFood(a.intendedFoodIntake())
-	if err != nil {
-		switch err.(type) {
-		case *infra.FloorError:
-		case *infra.NegFoodError:
-		case *infra.AlreadyEatenError:
-		default:
-		}
-	} else {
+	intendedFood := a.intendedFoodIntake()
+	foodTaken, err := a.TakeFood(intendedFood)
+
+	// if err != nil {
+	// 	switch err.(type) {
+	// 	case *infra.FloorError:
+	// 	case *infra.NegFoodError:
+	// 	case *infra.AlreadyEatenError:
+	// 	default:
+	// 	}
+	// } else {
+	// 	a.lastFoodTaken = foodTaken
+	// 	// Exponential moving average filter to average food taken whilst discounting previous food
+	// 	a.updateAverageIntake(foodTaken)
+	// }
+
+	if err == nil {
 		a.lastFoodTaken = foodTaken
+		// Exponential moving average filter to average food taken whilst discounting previous food
+		a.updateAverageIntake(foodTaken)
+		// Updates trust in the above neighbour based on average food
+		if a.averageFoodIntake < 1 {
+			a.updateTrust(-1, a.neighbours.above)
+		} else {
+			a.updateTrust(1, a.neighbours.above)
+		}
+		a.Log("Agent6 took food!")
 	}
 
 	// Reset the reqLeaveFoodAmount to nothing once the agent has eaten
 	if a.HasEaten() {
 		a.reqLeaveFoodAmount = -1
 	}
 
-	// Exponential moving average filter to average food taken whilst discounting previous food
-	a.updateAverageIntake(foodTaken)
-
 	// LOG
-	// a.Log("Team 6 agent has floor:", infra.Fields{"floor": a.Floor()})
-	// a.Log("Team 6 agent has HP:", infra.Fields{"hp": a.HP()})
-	// a.Log("Team 6 agent desired to take:", infra.Fields{"desiredFood": a.desiredFoodIntake()})
-	// a.Log("Team 6 agent intended to take:", infra.Fields{"intendedFood": a.intendedFoodIntake()})
-	// a.Log("Team 6 agent took:", infra.Fields{"foodTaken": foodTaken, "bType": a.currBehaviour.String()})
-
-	//fmt.Println(a.ActiveTreaties())
+	a.Log("Team 6 agent has floor:", infra.Fields{"floor": a.Floor()})
+	a.Log("Team 6 agent has HP:", infra.Fields{"hp": a.HP()})
+	a.Log("Team 6 agent desired to take:", infra.Fields{"desiredFood": a.desiredFoodIntake()})
+	a.Log("Team 6 agent intended to take:", infra.Fields{"intendedFood": a.intendedFoodIntake()})
+	a.Log("Team 6 agent took:", infra.Fields{"foodTaken": foodTaken, "bType": a.currBehaviour.string()})
```

Comment:
can you add all these to one log so that the logs file doesn't become extremely large